packages/vite-plugin-svelte/src/utils/vite-plugin-svelte-cache.ts

```diff
@@ -124,4 +140,43 @@ export class VitePluginSvelteCache {
 	private _getResolvedSvelteFieldKey(importee: string, importer?: string): string {
 		return importer ? `${importer} > ${importee}` : importee;
 	}
+
+	public async getPackageInfo(file: string): Promise<PackageInfo> {
+		let info = this._packageInfos.find((pi) => file.startsWith(pi.path));
+		if (!info) {
+			info = await findPackageInfo(file);
+			this._packageInfos.push(info);
+		}
+		return info;
+	}
+}
+
+/**
+ * utility to get some info from the package.json a file belongs to (closest one with a "name" set)
```

Comment:
could simplify this description:
```suggestion
 * utility to get some info from the closest package.json with a "name" set
```